src/components/Badge/Badge.st.css

```diff
@@ -0,0 +1,60 @@
+:import {
+    -st-from: "../../common/formatters.st";
+    -st-named: color, applyOpacity, fallback;
+}
+
+:vars {
+    /*Defaults*/
+    DefaultBgColor: color-5;
+    DefaultTextColor: color-1;
+    LightBgColor: color-1;
+    LightTextColor: color-5;
+    LightBorderColor: color-5;
+    PrimaryBgColor: color-8;
+    PrimaryTextColor: color-1;
+}
+
+:vars {
+    /*Overrides*/
+    BadgeBgColor: --overridable;
+    BadgeTextColor: --overridable;
+    BadgeBorderColor: --overridable;
+}
+
+:vars {
+    badgeBgColor: color(fallback(value(BadgeBgColor), value(DefaultBgColor)));
+    badgeTextColor: color(fallback(value(BadgeTextColor), value(DefaultTextColor)));
+    badgeBorderColor: color(fallback(value(BadgeBorderColor), transparent));
+}
+
+.root {
+    -st-states: priority(enum(default, light, primary));
+    display: inline-block;
+    border-radius: 12px;
+    padding: 0 12px;
+    line-height: 20px;
+    font-size: 12px;
+    border-width: 1px;
+    border-style: solid;
+    background-color: applyOpacity(value(badgeBgColor), 0.4);
```

Comment:
The opacity should be applied only on the preset colors.
When users override the colors, they should be able to choose whatever color and opacity they want.
I suggest moving the `applyOpacity` function to `vars` definition in lines 25-27.